### ISSUE TITLE
Applying blocks APIs

### ISSUE THREAD

**Issue by evanlinjin** [bitcoindevkit/bdk#903]
We need a set of APIs to easily apply blocks to a keychain tracker. I propse the following:

* `ChainGraph::<ConfirmationTime>::from_block(block)`: This creates a chain graph with two checkpoints (one for the block at height and one for prev block hash).
* `KeychainTracker::apply_update_relevant(scan: KeychainScan)`, does this same thing as `apply_update` but doesn't not take transactions that don't send to or from things in the txout index. Also add `determine_relevant_changeset`.
* `ChainGraph::apply_update_relevant(|tx| is_this_relevant(tx))` same as above except you can pass a predicate (also needs the changeset option).

~In addition to adding these APIs it looks like `KeychainTracker::determine_changeset` has a TODO in it that needs to be fixed before this whole workflow will work properly:~

Old ticket: https://github.com/LLFourn/bdk_core_staging/issues/193

**Comment by LLFourn** [comment]
After #895,  `ChainGraph::from_block` is no longer needed and `apply_update_relevent` applies to `IndexedTxGraph`.

**Comment by danielabrozzoni** [comment]
See https://github.com/bitcoindevkit/bdk/pull/896#issuecomment-1667437770

**Comment by LLFourn** [comment]
I think we can say this is done now. If disagree re-open. Improvements to it can be done in other issues.
